### ISSUE TITLE
Invalid field layout ID

### ISSUE THREAD

**Issue by circa1977** [angell-co/Spoon#84]
I just ran into Issue #79. So I removed the fields from the field layouts I had created and deleted the field layout tabs...and now my entire control panel is inaccessible.

![Screen Shot 2020-01-31 at 3 47 48 PM](https://user-images.githubusercontent.com/49657/73573325-1428c280-4441-11ea-86ee-bf25fa7b4572.png)

Craft 3.4.1.

**Comment by circa1977** [comment]
Fortunately I had a database backup from the last time I worked on the project and didn't lose anything. For a paid plugin that's only supposed to enable an organization layer, this is a pretty big concern. I was completely shut out of the control panel, and reviewing the database, that ID exists.

**Comment by circa1977** [comment]
As soon as I hit Install on the plugin again—which was uninstalled following the database backup restore, I get the error again.

If I can't rely on the plugin, I'd like to fully uninstall it, as there are still database tables intact from when it was installed.

Here's the full trace:

yii\base\InvalidConfigException: Invalid field layout ID: 30 in /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/src/behaviors/FieldLayoutBehavior.php:124
Stack trace:
#0 [internal function]: craft\behaviors\FieldLayoutBehavior->getFieldLayout()
#1 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/base/Component.php(297): call_user_func_array(Array, Array)
#2 /Users/markreeves/Sites/idg/codiak/craft/vendor/angellco/spoon/src/services/BlockTypes.php(534): yii\base\Component->__call('getFieldLayout', Array)
#3 /Users/markreeves/Sites/idg/codiak/craft/vendor/angellco/spoon/src/services/BlockTypes.php(166): angellco\spoon\services\BlockTypes->_populateBlockTypeFromRecord(Object(angellco\spoon\records\BlockType))
#4 /Users/markreeves/Sites/idg/codiak/craft/vendor/angellco/spoon/src/services/Loader.php(288): angellco\spoon\services\BlockTypes->getByContext('global', 'context')
#5 /Users/markreeves/Sites/idg/codiak/craft/vendor/angellco/spoon/src/services/Loader.php(246): angellco\spoon\services\Loader->fieldManipulator('global', false)
#6 /Users/markreeves/Sites/idg/codiak/craft/vendor/angellco/spoon/src/Spoon.php(96): angellco\spoon\services\Loader->run()
#7 [internal function]: angellco\spoon\Spoon->angellco\spoon\{closure}(Object(yii\base\Event))
#8 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/base/Event.php(312): call_user_func(Object(Closure), Object(yii\base\Event))
#9 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/base/Component.php(636): yii\base\Event::trigger('craft\\services\\...', 'afterLoadPlugin...', Object(yii\base\Event))
#10 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/src/services/Plugins.php(272): yii\base\Component->trigger('afterLoadPlugin...')
#11 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/src/base/ApplicationTrait.php(1351): craft\services\Plugins->loadPlugins()
#12 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/src/web/Application.php(103): craft\web\Application->_postInit()
#13 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/base/BaseObject.php(109): craft\web\Application->init()
#14 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/base/Application.php(206): yii\base\BaseObject->__construct(Array)
#15 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/src/web/Application.php(91): yii\base\Application->__construct(Array)
#16 [internal function]: craft\web\Application->__construct(Array)
#17 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/di/Container.php(400): ReflectionClass->newInstanceArgs(Array)
#18 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/di/Container.php(159): yii\di\Container->build('craft\\web\\Appli...', Array, Array)
#19 /Users/markreeves/Sites/idg/codiak/craft/vendor/yiisoft/yii2/BaseYii.php(365): yii\di\Container->get('craft\\web\\Appli...', Array, Array)
#20 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/bootstrap/bootstrap.php(245): yii\BaseYii::createObject(Array)
#21 /Users/markreeves/Sites/idg/codiak/craft/vendor/craftcms/cms/bootstrap/web.php(51): require('/Users/markreev...')
#22 /Users/markreeves/Sites/idg/codiak/craft/public/index.php(20): require('/Users/markreev...')
#23 {main}

**Comment by circa1977** [comment]
I also encountered #57 when I was removing the field layouts.

**Comment by circa1977** [comment]
After removing the plugin (see #85), `craft_spoon_blocktypes` has 5 records. 3 of those have `fieldLayoutId` values that are not `null`. They are:

* `30`
* `31`
* `32`

Table `craft_fieldlayouts` only goes up to id=`29`.

**Comment by joshangell** [comment]
Hi @circa1977, sorry you’re experiencing this issue.

Whilst I’m working on a fix for #79 you could try dropping that table entirely if you don’t mind losing the field configurations. It was an oversight on my part to not drop it when running uninstall - I will fix that during this next sprint.

If you want to resolve the error temporarily, I think what you’re running into can be solved by removing the fields and tabs from the field layouts like you did initially, then hitting save on the grouping view again.

Alternatively I would do a full restore from before the upgrade and pin your Craft version to 3.3.x until I get it working on 3.4.x

**Comment by circa1977** [comment]
> If you want to resolve the error temporarily, I think what you’re running into can be solved by removing the fields and tabs from the field layouts like you did initially, then hitting save on the grouping view again.

Per #79, if I enable the plugin, I can't access anything in the control panel. The moment I removed those fields, a disconnect emerged. Rolling back the database showed the plugin present, but not installed. If I install it, every control panel returns that error.

I had to remove the plugin. I've been purposely staying very lean on plugins in Craft to avoid conflicts and upgrade issues. Lesson learned, I think. I don't think I can justify this one.
